### PR TITLE
postgresql: 9.4 -> 9.5

### PR DESCRIPTION
**Pull request by teburd**
Updates postgresql to its latest version

### CODE REVIEW

**Comment by bfrog** [comment]
I'm new to contributing to nix here, and I'm looking for feedback on this, it doesn't entirely seem to work yet.

File: nixos/modules/services/databases/postgresql.nix [hunk 1]
```diff
@@ -158,7 +158,7 @@ in
       # Note: when changing the default, make it conditional on
       # ‘system.stateVersion’ to maintain compatibility with existing
       # systems!
```
Comment (domenkozar) [inline nixos/modules/services/databases/postgresql.nix]: This should probably depend on `system.stateVersion`

Comment (bfrog) [inline nixos/modules/services/databases/postgresql.nix]: I disabled updating the default for the service itself for now

**Comment by bfrog** [comment]
It works now as far as I can tell! I'm still a bit confused by the need for a separate top-level/all-applications.nix as I'm new, but the rest made sense to me

**Comment by spinus** [comment]
@bfrog I'm not sure what the convention here is but I wouldn't say it's upgrade, looks  like just adding 9.5 not upgrading/replaceing 9.4 -> 9.5

**Pull request merged**